### ISSUE TITLE
Patch Add operation on sub-attribute of a complex attribute not working as expected

### ISSUE THREAD

**Issue by jivpif** [simpleidserver/SimpleIdServer#67]
Hi,

I think there is an issue with the following case:

Given the data:

```
"roles": [
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Member",
        "display": "Member"
    }
]
```

When we execute any of the next Add operations:

```
"op": "Add",
"path": "roles.display",
"value": "Team Member"
```
or 
```
"op": "Add",
"path": "roles[value eq \"Member\"].display",
"value": "Team Member"
```

The result is:
```
"roles": [
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Member",       
    }
],
"display": "Team Member"
```

Expected:

```
"roles": [
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Member",  
        "display": "Team Member"    
    }
]
```
Scim patch Add operation reference : https://tools.ietf.org/html/rfc7644#section-3.5.2.1

> o  If the target location exists, the value is replaced.

Thanks

**Comment by simpleidserver** [comment]
Hi,

The issue should be fixed in the branch "release/1.1.9"

**Comment by jivpif** [comment]
Tried it. Looks good. Thanks : )
